cli/pcluster/numbers_to_list.py

```diff
@@ -0,0 +1,57 @@
+import copy
+
+
+class NumberedList:
+    """
+    This class adds number to a list of string and associate each one with one element of the list.
+
+    It can be used to help a user decide an option faster. It also accepts extra parameters to retrieve the elements.
+    """
+
+    def __init__(self, items):
+        """
+        Initialize the object with the list of string in which you want to add number.
+
+        Note that it can also be a tuple (item, extra-identifier1, ..., extra-identifierN). In this case, the item to
+        associate must be the first one of the tuple.
+        For example, you can pass to the function [["id1","synonym1"], ["id2","synonym2"]].
+        The get will retrieve: ["1",[id1","synonym1"], ["2",[id2","synonym2"]].
+        If you want to retrieve "id1" you could either use "1" or "synonym1" and for the "id2" you could either use "2"
+        or "synonym2"
+
+        :param items: the list of string in which number will be added.
+        """
+        self.numbered_items = []
+        self.param_to_items = {}
+        for number, element in enumerate(items, 1):
+            if isinstance(element, (list, tuple)):
+                value = element[0]
+                for synonym in element[1::]:  # Skip the first
+                    self.param_to_items[synonym] = value
+                self.param_to_items[number] = value
+            else:
+                self.param_to_items[number] = element
+            self.numbered_items.append((number, element))
+
+    def get(self):
+        """
+        Return an ordered list of the items and the extra-identifiers with the numbers added.
+
+        So for example if items was [["id1","synonym1"], ["id2","synonym2"]], the result list will return
+        ["1",[id1","synonym1"], ["2",[id2","synonym2"]]
+
+        :return: the ordered dict
+        """
+        return copy.copy(self.numbered_items)
+
+    def get_item_by(self, key):
```

Comment:
I don't like much that if you provide directly string that you are searching for, it will return a ValueError.